source/core/src/main/com/csse3200/game/inventory/Inventory.java

```diff
@@ -442,7 +338,14 @@ private Optional<Integer> getItemIndex(int code) {
      *         or {@link Optional#empty()} if the item is not in the inventory.
      */
     private Optional<Integer> getItemIndex(String name) {
-        return this.hasItem(name) ? Optional.of(nameToIndices.get(name).first()) : Optional.empty();
+        int i = 0;
+        for (AbstractItem item : memoryView) {
+            if (item.getName().equals(name)) {
+                return Optional.of(i);
+            }
+            i++;
+        }
+        return Optional.empty();
```

Comment:
Is there an advantage to using Optional instead of returning -1 for this case, similar to how you've done it in the add method?

Reply:
Nah, both are fine (Optional is probably more readable) - but I don't think it's being used anyway. I had it in in case someone wanted to do something such as checking whether the user has a certain item or number of items on them (e.g. a key to get through to the other area or smthn) but unless that materialises in the next 10 days the code is kind of pointless anyway.